### DISCUSSION TITLE
Having docker related files in a sub directory

### DISCUSSION THREAD

**Discussion by pascal-mueller** [microsoft/vscode-docker#3715] [Q&A]
Hello,

Assume `/` is my project root. Assume I put all my Dockerfiles and docker-compose files in `/deployment`. So if I wanted to build an image, I'd do:

```
cd /
docker build . -f deployment/SomeDockerfile -t my_app:latest
```

or something similar. Please note the `.` which tells Docker the context. Now I'd like to just rightclick the Dockerfile and/or use ctrl+shit+p to build the image.

Similar issue with docker compose. E.g. you also want to provide a `.env` file.

I was wondering, how you all solved this because I can't be the only one who does this, no?

**Comment by bwateratmsft** [comment]
Take a look at the [`docker-build` task](https://code.visualstudio.com/docs/containers/reference#_docker-build-task), I think it can do what you need. It allows you to separately specify the Dockerfile and context, along with many other arguments. There is a similar [`docker-compose` task](https://code.visualstudio.com/docs/containers/reference#_docker-compose-task) as well.

For example, do do the above `docker build` command, this task in `.vscode/tasks.json` should work (paths may need a little adjusting):

```json
		{
			"type": "docker-build",
			"label": "docker-build",
			"dockerBuild": {
				"tag": "my_app:latest",
				"dockerfile": "${workspaceFolder}/deployment/SomeDockerfile",
				"context": "${workspaceFolder}"
			}
		},
```

When you right-click > Build that Dockerfile, we look into `.vscode/tasks.json` to see if there is a build task that references it, and if so we execute the task instead of the default `docker build` terminal command.

**Reply by pascal-mueller** [reply]
Thanks, I'll give that a go :)
